tests/fallback_test.py

```diff
@@ -0,0 +1,210 @@
+            bootstrap=FallbackTest.bootstrap,
+            node_id=_NODE_ID.value,
+        )
+
+    def start_client(self, port: int = None, name: str = None):
+        logger.debug("Starting client process")
+        return framework.helpers.docker.Client(
+            manager=self.process_manager,
+            name=name or framework.xds_flags.CLIENT_NAME.value,
+            port=port or get_free_port(),
+            url="xds:///listener_0",
+            image=framework.xds_k8s_flags.CLIENT_IMAGE.value,
+        )
+
+    def start_control_plane(self, name: str, index: int, upstream_port: int):
+        logger.debug('Starting control plane "%s"', name)
+        return framework.helpers.docker.ControlPlane(
+            self.process_manager,
+            name=name,
+            port=self.bootstrap.xds_config_server_port(index),
+            upstream=f"{_HOST_NAME.value}:{upstream_port}",
+            image=_CONTROL_PLANE_IMAGE.value,
+        )
+
+    def start_server(self, name: str, port: int = None):
+        logger.debug('Starting server "%s"', name)
+        port = get_free_port() if port is None else port
+        return framework.helpers.docker.GrpcProcess(
+            name=name,
+            port=port,
+            image=framework.xds_k8s_flags.SERVER_IMAGE_CANONICAL.value,
+            manager=self.process_manager,
+            ports={8080: port},
+            command=[],
+        )
+
+    def test_fallback_on_startup(self):
+        with (
+            self.start_server(name="server1") as server1,
+            self.start_server(name="server2") as server2,
+            self.start_client() as client,
+        ):
+            self.assertTrue(
+                client.expect_message_in_output(
+                    "UNAVAILABLE: xDS channel for server"
+                )
+            )
+            self.assertEqual(client.get_stats(5).num_failures, 5)
+            # Fallback control plane start, send traffic to server2
+            with self.start_control_plane(
+                "fallback_xds_config", 1, server2.port
+            ):
+                stats = client.get_stats(5)
+                self.assertGreater(stats.rpcs_by_peer["server2"], 0)
+                self.assertNotIn("server1", stats.rpcs_by_peer)
+                # Primary control plane start. Will use it
+                with self.start_control_plane(
+                    name="primary_xds_config",
+                    index=0,
+                    upstream_port=server1.port,
+                ):
+                    self.assertTrue(
+                        client.expect_message_in_output(
+                            "parsed Cluster example_proxy_cluster", timeout_s=30
+                        )
+                    )
+                    stats = client.get_stats(10)
+                    self.assertEqual(stats.num_failures, 0)
+                    self.assertIn("server1", stats.rpcs_by_peer)
+                    self.assertGreater(stats.rpcs_by_peer["server1"], 0)
```

Comment:
Will all RPCs going to `server1` once it's up? If they are, we probably want to check no RPC is going to `server2` (same questions for all test cases).

Reply:
Some will still arrive to server2 while the primary config is being downloaded. Other checks verify that all RPCs go to `server1` so no need to check that none arrived to `server2`